### ISSUE TITLE
HDR not perfect on iOS

### ISSUE THREAD

**Issue by skrew** [mpv-player/mpv#8468]
### Important Information
- LIBMPV 0.33.0
- iOS 14.3

### Reproduction steps
Play a HDR content on iOS, it's better on libmpv 0.33 which have this patch: https://github.com/mpv-player/mpv/pull/7849/files but not "perfect"

### Expected behavior
HDR fully supported, like MPV on Mac

### Actual behavior
There are still glitches, as you can see on the sky on the screenshot

### Sample files
http://skrobs.com/hdrb.mkv

### Log file
[mpv-log.txt](https://github.com/mpv-player/mpv/files/5816273/mpv-log.txt)
![IMG_0234](https://user-images.githubusercontent.com/738170/104632617-435e3100-569e-11eb-8d4e-f61bada2395a.png)

**Comment by Akemi** [comment]
pinging @tmm1 because he might know more. problem is the banding in the sky.

**Comment by tmm1** [comment]
AFAIK the only way to get proper HDR rendering on iOS is via AVFoundation. I have tried a million things with both opengl and metal and it never renders correctly.

**Comment by Akemi** [comment]
i believe this issue isn't really about HDR itself but HDR material (and probably other 10bit?) is affected by that banding issue.

**Comment by tmm1** [comment]
You could try this:

```diff
diff --git a/libavcodec/videotoolbox.c b/libavcodec/videotoolbox.c
index 49e726a75f..85a1de548f 100644
--- a/libavcodec/videotoolbox.c
+++ b/libavcodec/videotoolbox.c
@@ -1028,7 +1028,12 @@ static enum AVPixelFormat videotoolbox_best_pixel_format(AVCodecContext *avctx)

     int depth = descriptor->comp[0].depth;
     if (depth > 8) {
+#if TARGET_OS_IPHONE
+        /* iOS doesn't support 10 bit textures in GLES */
+        return AV_PIX_FMT_BGRA;
+#else
         return AV_PIX_FMT_P010;
+#endif
     }

     return AV_PIX_FMT_NV12;
```

**Comment by skrew** [comment]
Well I need to sleep !

First, I saw that I had FFMPEG 4.1.3 and NOT 4.3.1!

So I built 4.3.1 version without the patch and the result looks perfect (look the screenshot)
Just in case, I applied the patch and the result is the same: perfect. Maybe the patch can be useful in some cases, I've only tested with 3 files, so I'll keep it (thanks !).

So I'm sorry I took your time when all I had to do was to update, which I normally do before opening an issue but since version 0.33 didn't work on my app, I couldn't test before.

Small consolation: I compared the same video on iOS between VLC and LIBMPV and the result is much better with LIBMPV ! The video from VLC is darker and have less contrast.

To summarize for those who are wondering: FFmpeg 4.3.1 (with @tmm1 patch ?) + LIBMPV 0.33 to have a working HDR on iOS.

I don't close the ticket yet because I still have to try with tvOS 🤞🏻

![IMG_0240](https://user-images.githubusercontent.com/738170/104671088-5fcd8e00-56dd-11eb-890c-a5147b7d13c8.png)

**Comment by skrew** [comment]
Bad news .... Yes, i will sleep now...

I had set `hwdec` to "no" while testing some hours ago... After set `hwdec` to "auto", the result is still bad... 

It's 03:27 AM here, sleeping ! i will continue testing tomorrow

**Comment by Akemi** [comment]
try `--hwdec=auto-copy` or `--hwdec=videotoolbox-copy`. that is mostly likely the issue mentioned in #7846 (and i mentioned here https://github.com/mpv-player/mpv/issues/8467#issuecomment-759901714), because there was no 16-bit UNORM for p010 texture format on iOS, i believe. if that's is still the case there isn't much we can do about?

**Comment by skrew** [comment]
Yes i think it's over until we get metal support or a wrapper supporting metal...

Thanks again.

**Comment by MenThu** [comment]
Hello, I am also trying to use libmpv to play videos on the iOS platform, but I have not seen the iOS Demo in this repository or on the entire github.
Seeing that you already have an iOS application based on libmpv, can you show some code on how to use libmpv on iOS 
especially the OpenGL ES rendering.

Thanks
